snippets/olmv1-operator-conditions-support.adoc

```diff
@@ -0,0 +1,12 @@
+// Text snippet included in the following modules:
+//
+// * modules/olmv1-supported-extensions.adoc
+// * release_notes/ocp-4-17-release-notes.adoc (enterprise-4.17 branch only)
+// * release_notes/ocp-4-16-release-notes.adoc (enteprise-4.16 branch only)
+// * release_notes/ocp-4-15-release-notes.adoc (enteprise-4.15 branch only)
+
+:_mod-docs-content-type: SNIPPET
+
+{olmv1-first} does not support the `OperatorConditions` API introduced in {olmv0}.
+
+Currently, there are no tests implemented that check for the use of this API. If an extension uses the `OperatorConditions` API, the extension does not install correctly. Most extensions that rely on this API fail at start time, but some might fail during reconciliation.
```

Comment:
This somewhat implies that we just haven't gotten around to implementing the test for the use of this API, but the design of the OperatorConditions API in OLMv0 makes it impossible for OLMv1 to know whether a cluster extension will use this API in advance. _That's_ why the test doesn't exist.

Not sure we want to go into that level of detail in the docs, but the point is that I don't think OLM will ever be able to do this check.

Reply:
Thanks for the clarification, Joe. I'll take another crack at it.

Reply:
@joelanford do we have a more customer-ready explanation why we cannot perform the tests? Do we have any alternative workarounds? I'm concerned customers will see this as a regression if we call it out with no explanation, and will file support tickets and get unsatisfying resolutions if we don't call it out at all.

Reply:
We should just remove the statement `there are no tests implemented that check for the use of this API` . I do not see this sentence adding any value.

Reply:
@mwerner2113 This API is not directly customer-facing API. It is an API that operator authors use to communicate back to OLMv0, mostly (entirely, perhaps?) for the purpose of preventing OLM from upgrading the operator during a critical section of code (e.g. a migration).

However, that feature gives authors a false sense of security and promotes poor practices. Other controllers or external factors could interrupt an OLM-managed operator during its critical section. Examples include:
   - horizontal pod autoscaler scaling down a deployment replica
   - machine config operator draining a node to perform an upgrade
   - network partition that causes cascading failures
   - power interruptions
   - container OOMKilled due to exceeding memory limit

Extension authors _should_ already be designing their extensions to have resilience to these interruptions. And if they do, `OperatorConditions` doesn't really buy them much.

In OLMv1, I suppose the workaround instructions would be:
- pin the operator to a specific version to avoid automatic upgrades
- consult the operator documentation to understand when it is safe to un- (or re-)pin the operator to induce an upgrade by OLM.